packages/solana/src/connectors/walletAdapters.ts

```diff
@@ -1,49 +1,20 @@
-import {
-  PhantomWalletAdapter,
-  BackpackWalletAdapter,
-  SolflareWalletAdapter,
-  TrustWalletAdapter
-} from './walletAdapters/index.js'
-
-import { WalletAdapterNetwork, type BaseWalletAdapter } from '@solana/wallet-adapter-base'
-import { solana, solanaDevnet, solanaTestnet } from '../utils/chains.js'
 import type { Connector } from '@web3modal/scaffold'
 import { ConstantsUtil } from '@web3modal/common'
 
-export type AdapterKey = 'phantom' | 'solflare' | 'trustWallet' | 'backpack'
-export const supportedWallets: AdapterKey[] = ['phantom', 'solflare', 'trustWallet', 'backpack']
-
-const chainMap = {
-  [solana.chainId]: WalletAdapterNetwork.Mainnet,
-  [solanaDevnet.chainId]: WalletAdapterNetwork.Devnet,
-  [solanaTestnet.chainId]: WalletAdapterNetwork.Testnet
-}
+import type { BaseWalletAdapter } from '@solana/wallet-adapter-base'
 
-export function createWalletAdapters(chainId?: string) {
-  return {
-    phantom: new PhantomWalletAdapter(),
-    trustWallet: new TrustWalletAdapter(),
-    backpack: new BackpackWalletAdapter(),
-    solflare: new SolflareWalletAdapter({
-      network: chainMap[chainId || solana.chainId]
-    })
-  }
-}
-
-export function syncInjectedWallets(
-  w3mConnectors: Connector[],
-  adapters: Record<AdapterKey, BaseWalletAdapter>
-) {
-  supportedWallets.forEach(wallet => {
-    if (window[wallet as keyof Window]) {
+export function syncInjectedWallets(w3mConnectors: Connector[], adapters: BaseWalletAdapter[]) {
+  for (const adapter of adapters) {
+    const name = adapter.name.toLocaleLowerCase() as keyof Window
+    if (window[name]) {
```

Comment:
Let's remove this conditional, all the adapters should show up on the modal even when not installed.